webapp/src/app/admin/devices/devices.component.ts

```diff
@@ -83,6 +91,66 @@ export class DevicesComponent implements OnInit {
             }
         });
     }
+
+    async addRecovery() {
+        const accounts: IAccount[] = [];
+        Log.print("updating uData");
+        await this.uData.contact.updateOrConnect(this.uData.bc, true);
+        Log.print(this.uData.contacts);
+        await showSnack(this.snack, "Searching actions and groups", async () => {
+            for (const c of this.uData.contacts) {
+                accounts.push({
+                    id: (await c.getDarcSignIdentity()).id,
+                    name: c.alias,
+                });
+            }
+            const darcs = (await this.uData.contact.getGroups())
+                .concat(await this.uData.contact.getActions());
+            for (const g of darcs) {
+                accounts.push({
+                    id: g.darc.getBaseID(),
+                    name: g.darc.description.toString(),
+                });
+            }
+        });
+        if (accounts.length === 0) {
+            return showDialogInfo(this.dialog, "No accounts found",
+                "Sorry, there are no accounts linked to this user.", "OK");
+        }
+        const ac = this.dialog.open(DeviceRecoveryComponent, {data: accounts});
+        ac.afterClosed().subscribe(async (result) => {
+            if (result) {
+                const d = accounts.find((acc) => acc.id.equals(result));
+                if (d == null) {
```

Comment:
`null` isn't returned by find (or you have a list with some `null` value).
```suggestion
                if (d == undefined) {
```